File: modules/jooby-apt/src/test/java/tests/i3476/C3476.java
/*
 * Jooby https://jooby.io
 * Apache License Version 2.0 https://jooby.io/LICENSE.txt
 * Copyright 2014 Edgar Espina
 */
package tests.i3476;

import java.util.List;

import io.jooby.annotation.GET;
import io.jooby.annotation.QueryParam;

public class C3476 {

  @GET("/3476")
  public <T> List<T> get(@QueryParam List<T> list) {
    return List.of();
  }

  @GET("/box")
  public Box<Integer> box(@QueryParam Box<Integer> box) {
    return box;
  }
}

File: tests/src/test/kotlin/i3476/C3476.kt
/*
 * Jooby https://jooby.io
 * Apache License Version 2.0 https://jooby.io/LICENSE.txt
 * Copyright 2014 Edgar Espina
 */
package i3476

import io.jooby.annotation.GET
import io.jooby.annotation.QueryParam

class C3476 {
  @GET("/3476") fun <E> shouldGenerateGenerics(@QueryParam bean: List<E>): List<E> = listOf()
}
